naga/src/back/hlsl/writer.rs

```diff
@@ -2579,7 +2583,7 @@ impl<'a, W: fmt::Write> super::Writer<'a, W> {
                 crate::Literal::F64(value) => write!(self.out, "{value:?}L")?,
                 crate::Literal::F32(value) => write!(self.out, "{value:?}")?,
                 crate::Literal::U32(value) => write!(self.out, "{value}u")?,
-                crate::Literal::I32(value) => write!(self.out, "{value}")?,
+                crate::Literal::I32(value) => write!(self.out, "int({value})")?,
```

Comment:
Why is this necessary? This should have a comment, at least.

Reply:
done